numba/tests/test_builtins.py

```diff
@@ -948,6 +1030,64 @@ def test_sum_exceptions(self):
 
         self.assertIn('No implementation', str(raises.exception))
 
+    def test_isinstance(self):
+        pyfunc = isinstance_usecase
+        cfunc = jit(nopython=True)(pyfunc)
+
+        inputs = (
+            3,              # int
+            5.0,            # float
+            "Hello",        # string
+            b'world',       # bytes
+            1j,             # complex
+            [1, 2, 3],      # list
+            (1, 3, 3, 3),   # UniTuple
+            set([1, 2]),    # set
+            (1, 'nba', 2),  # Heterogeneous Tuple
+            # {'hello': 2},   # dict - doesn't work as input
+        )
+
+        for inpt in inputs:
+            expected = pyfunc(inpt)
+            got = cfunc(inpt)
+            self.assertEqual(expected, got)
+
+    def test_isinstance_numba_types(self):
+        pyfunc = isinstance_usecase_numba_types
+        cfunc = jit(nopython=True)(pyfunc)
+
+        inputs = (
+            (types.int32(1), 'int32'),
+            (types.int64(2), 'int64'),
+            (types.float32(3.0), 'float32'),
+            (types.float64(4.0), 'float64'),
+            (types.complex64(5j), 'no match'),
+            (typed.List([1, 2]), 'typed list'),
```

Comment:
`typed.Dict`?

Reply:
Also, should this sort of thing work?
```python
from numba import njit
import numpy as np

@njit
def foo():
    a = {'a': 10, 'b': np.zeros(3)}
    if isinstance(a, dict):
        print(True)
    else:
        print(False)

print(foo())
```
?

Reply:
Done!